poetry/repositories/pypi_repository.py

```diff
@@ -421,11 +423,72 @@ def _get_info_from_urls(
                 return info
 
             if platform_specific_wheels and "sdist" not in urls:
-                # Pick the first wheel available and hope for the best
-                return self._get_info_from_wheel(platform_specific_wheels[0])
+                # Attempt to select the best platform-specific wheel
+                best_wheel = self._pick_platform_specific_wheel(
+                    platform_specific_wheels
+                )
+                return self._get_info_from_wheel(best_wheel)
 
         return self._get_info_from_sdist(urls["sdist"][0])
 
+    def get_sys_info(self):  # type: () -> Dict[str, str]
+        # Return system information. Can be overridden for testing
+        return {
+            "plat": platform.system().lower(),
+            "is32bit": sys.maxsize <= 2 ** 32,
+            "pyver": platform.python_version_tuple(),
+        }
+
+    def _pick_platform_specific_wheel(
+        self, platform_specific_wheels
+    ):  # type: (list) -> str
+        sys_info = self.get_sys_info()
+        # Format the expected platform name as used by package authors
+        os_map = {"windows": "win", "darwin": "macosx"}
+        os_name = (
+            os_map[sys_info["plat"]] if sys_info["plat"] in os_map else sys_info["plat"]
+        )
+        bit_label = "32" if sys_info["is32bit"] else "64"
+        py_label = "cp{}".format("".join(sys_info["pyver"][:2]))
+        self._log(
+            "Attempting to determine best wheel file for: {}".format(sys_info),
+            level="debug",
+        )
+
+        platform_matches = []
+        for url in platform_specific_wheels:
+            m = wheel_file_re.match(Link(url).filename)
```

Comment:
Uses the `wheel_file_re` regex to parse the wheel filename for the PEP425 compatibility tags